host/sgxload.c

```diff
@@ -781,27 +746,24 @@ oe_result_t oe_sgx_delete_enclave(oe_enclave_t* enclave)
     }
     else /* FLC simulation mode needs to munmap. */
 #endif
+
+#endif /* END __LINUX __*/
     {
-        /* Non-FLC Linux and simulation mode both allocate memory. */
-        munmap((void*)enclave->addr, enclave->size);
+        /* free allocate memory. */
+        result = _sgx_free_enclave_memory((void*)enclave->addr, enclave->size);
```

Comment:
This is an incorrect refactor; the point of creating `_sgx_free_enclave_memory` is to consolidate all of the different deallocation build logic into a single function, not that `_sgx_free_enclave_memory` replaces `munmap` directly. i.e. lines 772-800 should be the implementation of `_sgx_free_enclave_memory` (except that line 796 replaces all of 790-799)

This is also likely what is causing tests to fail in the SGX1FLC configuration in your CI runs.